### PR TITLE
ADD Soft Precision and Recall metrics

### PR DESCRIPTION
**Pull request by Jad-yehya**

### CODE REVIEW

File: benchmark_utils/__init__.py [hunk 1]
```diff
@@ -165,7 +133,7 @@ def soft_recall(y_true: np.ndarray,
             fa += 1
 
         if y_true[i] == 1 and (
-                y_true[left:right] == 1).any():
+                y_pred[left:right] == 1).any():
```
Comment (tomMoral) [inline benchmark_utils/__init__.py:136]: Put these metrics in a `metrics.py` file.

File: benchmark_utils/__init__.py [hunk 2]
```diff
@@ -96,7 +64,7 @@ def soft_precision2(y_true: np.ndarray,
             ma += 1
```
Comment (tomMoral) [inline benchmark_utils/__init__.py:64]: You can do better with vectorial operations: 
1. do a convolution of `y_true` with a patch of one of size `2*detection_range`, 
2. reshape it and multiply it with `y_pred`,
3. do the sum and divide by the sum in `y_true`.

**Review by tomMoral — COMMENTED** [review]
Some comments to improve the code.

File: benchmark_utils/__init__.py (hunk 1 above)
Comment (Jad-yehya) [reply benchmark_utils/__init__.py:136]: Done.

**Pull request merged**